express-error-handling/server.ts

```diff
@@ -0,0 +1,119 @@
+import express from 'express';
+import { errorMiddleware, ClientError } from './lib/index.js';
+
+type Note = {
+  noteId: number;
+  content: string;
+};
+
+const app = express();
+
+app.get('/api/notes', async (req, res, next) => {
+  try {
+    const notes = await readNotes();
+    res.send(notes);
+  } catch (err) {
+    console.error(err);
```

Comment:
You should _replace_ the `console.error` and `res.status` lines with the call to `next(err);`. Instead, you just added the call to `next(err);`. Delete the other two lines. Here and below.